internal/versions/versions.go

```diff
@@ -92,37 +121,238 @@ func (h *handler) GetMustGatherImages(openshiftVersion, cpuArchitecture, pullSec
 
-func (h *handler) GetReleaseImage(ctx context.Context, openshiftVersion, cpuArchitecture, pullSecret string) (*models.ReleaseImage, error) {
-	image, err := h.getReleaseImageFromCache(openshiftVersion, cpuArchitecture)
-	if err == nil || h.kubeClient == nil {
-		return image, err
+func (H *handler) getURLForDBRelease(release common.OpenshiftVersion, cpuArchitecture string) *string {
+	mapToURLCpuArchitecture := map[string]string{
+		common.AMD64CPUArchitecture: common.X86CPUArchitecture,
+		common.ARM64CPUArchitecture: common.AARCH64CPUArchitecture,
 	}
 
-	// The image doesn't exist in the cache.
-	// Fetch all the cluster image sets, cache them, then search the cache again
+	arch, ok := mapToURLCpuArchitecture[cpuArchitecture]
+	if !ok {
+		arch = cpuArchitecture
+	}
 
+	return swag.String(fmt.Sprintf(releaseImageURLPrefix+"%s-%s", *release.DisplayName, arch))
+}
+
+func (h *handler) getReleaseSupportLevel(release models.ReleaseImage) (*string, error) {
+	if release.SupportLevel != "" {
+		return &release.SupportLevel, nil
+	}
+
+	version := strings.TrimSuffix(*release.Version, common.MultiCPUArchitecture)
+
+	isPreRelease, err := common.IsVersionPreRelease(version)
+	if err != nil {
+		return nil, err
+	}
+
+	if *isPreRelease {
+		return swag.String(models.OpenshiftVersionSupportLevelBeta), nil
+	}
+
+	majorMinorVersion, err := common.GetMajorMinorVersion(version)
+	if err != nil {
+		return nil, err
+	}
+
+	supportLevels, err := common.GetOpenshiftVersionsSupportLevelFromDBWhere(h.db, "openshift_version = ?", majorMinorVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(supportLevels) == 0 {
+		return nil, errors.Errorf("error occurred while trying to get the support level of %s from DB", *release.Version)
+	}
+
+	return swag.String(supportLevels[0].SupportLevel), nil
+}
+
+func (h *handler) createReleaseImageFromDBOpenshiftVersion(
+	dbVersion *common.OpenshiftVersion,
+	cpuArch string,
+) (*models.ReleaseImage, error) {
+	if dbVersion == nil {
+		return nil, nil
+	}
+
+	releaseModel := models.ReleaseImage{
+		CPUArchitecture:  &cpuArch,
+		CPUArchitectures: []string{cpuArch},
+		Default:          dbVersion.Default,
+		OpenshiftVersion: dbVersion.DisplayName,
+		Version:          dbVersion.DisplayName,
+		SupportLevel:     "",
+		URL:              h.getURLForDBRelease(*dbVersion, cpuArch),
+	}
+
+	if cpuArch == common.MultiCPUArchitecture {
+		releaseModel.CPUArchitectures = supportedMultiArchitectures
+	}
+
+	if dbVersion.SupportLevel != nil {
+		releaseModel.SupportLevel = *dbVersion.SupportLevel
+		return &releaseModel, nil
+	}
+
+	newSuppotLevel, err := h.getReleaseSupportLevel(releaseModel)
+	if err != nil {
+		h.log.WithError(err).Debug("error occurred while trying to get openshift releases support levels")
+		return nil, err
+	}
+	releaseModel.SupportLevel = *newSuppotLevel
+
+	return &releaseModel, nil
+}
+
+func (h *handler) getReleaseImageFromDB(
+	openshiftVersion,
+	cpuArchitecture string,
+	versionSegmentsLength int,
```

Comment:
Maybe create named types for the options. Looking at the caller it's really hard to tell what this param is.